tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -55,80 +57,155 @@ def rotate(images, angles, interpolation="NEAREST"):
-                                    (image_height - 1))) / 2.0
-  num_angles = array_ops.shape(angles)[0]
-  return array_ops.concat(
-      values=[
-          math_ops.cos(angles)[:, None],
-          -math_ops.sin(angles)[:, None],
-          x_offset[:, None],
-          math_ops.sin(angles)[:, None],
-          math_ops.cos(angles)[:, None],
-          y_offset[:, None],
-          array_ops.zeros((num_angles, 2), dtypes.float32),
-      ],
-      axis=1)
-
-
-def transform(images, transforms, interpolation="NEAREST"):
+  with ops.name_scope(name):
+    angle_or_angles = ops.convert_to_tensor(
+        angles, name="angles", dtype=dtypes.float32)
+    if len(angle_or_angles.get_shape()) == 0:  # pylint: disable=g-explicit-length-test
+      angles = angle_or_angles[None]
+    elif len(angle_or_angles.get_shape()) == 1:
+      angles = angle_or_angles
+    else:
+      raise TypeError("Angles should have rank 0 or 1.")
+    x_offset = ((image_width - 1) - (math_ops.cos(angles) *
+                                     (image_width - 1) - math_ops.sin(angles) *
+                                     (image_height - 1))) / 2.0
+    y_offset = ((image_height - 1) - (math_ops.sin(angles) *
+                                      (image_width - 1) + math_ops.cos(angles) *
+                                      (image_height - 1))) / 2.0
+    num_angles = array_ops.shape(angles)[0]
+    return array_ops.concat(
+        values=[
+            math_ops.cos(angles)[:, None],
+            -math_ops.sin(angles)[:, None],
+            x_offset[:, None],
+            math_ops.sin(angles)[:, None],
+            math_ops.cos(angles)[:, None],
+            y_offset[:, None],
+            array_ops.zeros((num_angles, 2), dtypes.float32),
+        ],
+        axis=1)
+
+
+def translations_to_projective_transforms(
+    translations, name="translations_to_projective_transforms"):
+  """Returns projective transform(s) for the given translation(s).
+
+  Args:
+      translations: A 2-element list representing [dx, dy] or a vector of
+          2-element lists representing [dx, dy] to translate for each image
+          (for a batch of images). The rank must be statically known (the shape
+          is not `TensorShape(None)`.
+      name: The name of the op.
+
+  Returns:
+      A tensor of shape (num_images, 8) projective transforms which can be given
+          to `tf.contrib.image.transform`.
+  """
+  with ops.name_scope(name):
+    translation_or_translations = ops.convert_to_tensor(
+        translations, name="translations", dtype=dtypes.float32)
+    if len(translation_or_translations.get_shape()) == 1:
+      translations = translation_or_translations[None]
+    elif len(translation_or_translations.get_shape()) == 2:
+      translations = translation_or_translations
+    else:
+      raise TypeError("Translations should have rank 1 or 2.")
+    num_translations = array_ops.shape(translations)[0]
+    return array_ops.concat(
```

Comment:
Explain what you're doing here. Match dtype to image dtype?

Reply:
Actually, this just processes the translation matrices, which are always float32. Added a comment to clarify this, and describe the translation matrix.